runtime/lua/vim/health.lua

```diff
@@ -106,6 +106,8 @@
 local M = {}
 
 local s_output = {} ---@type string[]
+---@type { ['ok']: number?, ['warn']: number?, ['error']: number? }
+local check_summary = {}
```

Comment:
wouldn't need the `@type` if you define the table here. also eliminates the `or 0` in `M.ok()` etc.

```suggestion
local check_summary = {
  ok = 0,
  ...
}
```